src/object_accessor.hpp

```diff
@@ -98,37 +98,76 @@ void Object::set_property_value_impl(ContextType& ctx, const Property &property,
     switch (property.type & ~PropertyType::Nullable) {
         case PropertyType::Object: {
             ContextType child_ctx(ctx, property);
-            auto link = child_ctx.template unbox<RowExpr>(value, true, try_update);
-            table.set_link(col, row, link.get_index(), is_default);
+            // Check if we are linking to an object already
+            auto curr_link = table.get_link(col,row);
+            auto link = child_ctx.template unbox<RowExpr>(value, true, try_update, update_only_diff);
+            if (!update_only_diff || curr_link != link.get_index()) {
+                table.set_link(col, row, link.get_index(), is_default);
+                ctx.did_change();
```

Comment:
`ctx.will_change()` and `ctx.did_change()` calls always need to be paired. Either this needs to only call `will_change()` if we're actually setting the value or it needs to call `did_change()` even if we didn't.

Reply:
OK. I was not aware of that. I will just call `did_change()` unconditionally.

Reply:
Do we have or should we have tests that would catch this?

Reply:
This functionality is only tested at the Cocoa level since it's there to support KVO. Part of exposing this in the Cocoa binding will be to extend the existing KVO tests to use the new option.